crates/core/src/sync_local.rs

```diff
@@ -61,16 +89,22 @@ pub fn sync_local(db: *mut sqlite::sqlite3, _data: &str) -> Result<i64, SQLiteEr
             "\
 -- 1. Filter oplog by the ops added but not applied yet (oplog b).
 --    SELECT DISTINCT / UNION is important for cases with many duplicate ids.
-WITH updated_rows AS (
-  SELECT DISTINCT b.row_type, b.row_id FROM ps_buckets AS buckets
-    CROSS JOIN ps_oplog AS b ON b.bucket = buckets.id
-  AND (b.op_id > buckets.last_applied_op)
-  UNION SELECT row_type, row_id FROM ps_updated_rows
-)
+WITH 
+  involved_buckets (id) AS (
+    SELECT id FROM ps_buckets WHERE ?1 IS NULL
+      OR name IN (SELECT value FROM json_each(json_extract(?1, '$.buckets')))
+  ),
+  updated_rows AS (
+    SELECT DISTINCT FALSE as local, b.row_type, b.row_id FROM ps_buckets AS buckets
+      CROSS JOIN ps_oplog AS b ON b.bucket = buckets.id AND (b.op_id > buckets.last_applied_op)
+      WHERE buckets.id IN (SELECT id FROM involved_buckets)
+    UNION SELECT TRUE, row_type, row_id FROM ps_updated_rows
```

Comment:
`ps_updated_rows` is currently used to track both:
1. Changes made locally (to allow reverting the changes if the server does not apply them).
2. Removed rows (since we don't track individual REMOVE operations in ps_oplog anymore).